client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-prefetch-images.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'react';
+
+/**
+ * Prefetches the images that can be bottleneck for the pattern rendering.
+ */
+export const usePrefetchImage = () => {
+	const prefetchImage = ( url: string ) => {
+		const link = document.createElement( 'link' );
+		link.rel = 'prefetch';
+		link.as = 'image';
+		link.href = url;
+		document.head.appendChild( link );
+	};
+
+	const removeLinks = ( url: string ) => {
+		const link = document.querySelector( `link[href="${ url }"]` );
```

Comment:
How about storing the created elements so we don't need to query from the DOM?

Reply:
Thank you for your suggestion! I'm done with this commit. https://github.com/Automattic/wp-calypso/pull/75054/commits/1e9cece69e64a049ad41a6dcd46daa6d5368b0fb